apps/src/lib/kits/maker/boards/microBit/LightSensor.js

```diff
@@ -0,0 +1,77 @@
+import {EventEmitter} from 'events';
+import {sensor_channels, roundToHundredth} from './MicroBitConstants';
+
+export default class LightSensor extends EventEmitter {
+  constructor(board) {
+    super();
+    this.state = {
+      threshold: 128,
+      rangeMin: 0,
+      rangeMax: 255,
+      currentReading: 0
+    };
+    this.board = board;
+    this.board.mb.addFirmataUpdateListener(() => {
+      // Only emit the data event when the value is above the threshold or the
+      // previous reading was above the threshold
+      if (
+        this.board.mb.analogChannel[sensor_channels.lightSensor] >=
+          this.state.threshold ||
+        this.state.currentReading >= this.state.threshold
+      ) {
+        this.emit('data');
```

Comment:
The spec says that this should fire every 50ms, but that doesn't look like what you're checking here. Can the spec be updated to describe the expected behavior?

Reply:
I thought I had read that the default sampling period was 50ms, so I didn't set it on any of the sensors. However, I was looking for the documentation where I read that and now can't find it! So I'm adding a change that sets the sample rate for our sensors. Thanks for checking this!

Reply:
Also, just to clarify your comment, if we set the sampling interval, then we won't need to check the timing in this function - we'll just need to check if it meets the other requirements for triggering an event.